WalletWasabi/Legal/LegalDocuments.cs

```diff
@@ -0,0 +1,116 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.Helpers;
+using WalletWasabi.Logging;
+using WalletWasabi.WebClients.Wasabi;
+
+namespace WalletWasabi.Legal
+{
+	public class LegalDocuments
+	{
+		public const string EmbeddedFileName = "LegalDocuments.txt";
+		public const string LegalFolderName = "Legal";
+		public const string AssetsFoldername = "Assets";
+		public static readonly string EmbeddedFilePath = Path.Combine(EnvironmentHelpers.GetFullBaseDirectory(), LegalFolderName, AssetsFoldername, EmbeddedFileName);
+		public string FilePath { get; }
+		public Version Version { get; }
+
+		public LegalDocuments(Version version, string filePath)
+		{
+			Version = Guard.NotNull(nameof(version), version);
+			FilePath = Guard.NotNullOrEmptyOrWhitespace(nameof(filePath), filePath);
+		}
+
+		public static async Task<LegalDocuments> TryLoadAgreedAsync(string dataDir)
+		{
+			var legalFolderPath = Path.Combine(dataDir, LegalFolderName);
+			IoHelpers.EnsureDirectoryExists(legalFolderPath);
+
+			int fileCount = Directory.EnumerateFileSystemEntries(legalFolderPath).Count();
+			// If more than one file found, then something strange happened, delete the dir and start from zero.
+			if (fileCount > 1)
+			{
+				Logger.LogInfo("Multiple legal docs detected. Recovering empty legal directory...");
+				await IoHelpers.DeleteRecursivelyWithMagicDustAsync(legalFolderPath).ConfigureAwait(false);
+				IoHelpers.EnsureDirectoryExists(legalFolderPath);
+			}
+
+			await TryEnsureBackwardsCompatibilityAsync(dataDir).ConfigureAwait(false);
```

Comment:
I would ensure the backward compatibility first by checking if the Legal folder exists. If it is not there and there are wallets I would agree automatically. So move up this before line: 
https://github.com/zkSNACKs/WalletWasabi/pull/2805/files#diff-f8f98bd77ed196af990e22dd718ff036R33

Reply:
Cannot, the previous if may delete the folder, so backwards compatibility ensurment must be done after.

Reply:
I don't know if this is a special case but we used to say that what goes into the shared WalletWasabi project shouldn't use `ConfigureAwait(false)`

Reply:
@lontivero The exact opposite. Library project should always configure await. We used to say nothing has to configure await because in .NET Core it's irrelevant. It turns out it is only irrelevant in ASP.NET Core, so we decided the library will configure await.